src/v/kafka/requests/fetch_request.cc

```diff
@@ -340,6 +365,16 @@ read_from_ntp(op_context& octx, model::ntp ntp, fetch_config config) {
               return make_ready_partition_response_error(
                 error_code::not_leader_for_partition);
           }
+          if (mntpv.is_materialized()) {
+              if (auto log = mgr.log(mntpv.input_ntp())) {
```

Comment:
maybe we could have a method like `make_partition_wrapper` that would create a `partition_wrapper`. This would allow us to handle reads in the same way.

Reply:
agree. this should be unified if possible. i'm making a ticket to capture this clean-up.